packages/client/src/derivedProperties/createWithPropertiesObjectSet.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  ObjectOrInterfaceDefinition,
+  WithPropertyObjectSet,
+} from "@osdk/api";
+import type {
+  DerivedPropertyDefinition,
+  ObjectSet as WireObjectSet,
+  SelectedPropertyOperation,
+} from "@osdk/internal.foundry.core";
+import invariant from "tiny-invariant";
+import { modernToLegacyWhereClause } from "../internal/conversions/modernToLegacyWhereClause.js";
+
+let idCounter = 0;
+
+/** @internal */
+export function createWithPropertiesObjectSet<
+  Q extends ObjectOrInterfaceDefinition,
+>(
+  objectType: Q,
+  objectSet: WireObjectSet,
+  definitionMap: Map<any, DerivedPropertyDefinition>,
+): WithPropertyObjectSet<Q> {
+  const base: WithPropertyObjectSet<Q> = {
+    pivotTo: (link) => {
+      return createWithPropertiesObjectSet(objectType, {
+        type: "searchAround",
+        objectSet,
+        link,
+      }, definitionMap) as any;
```

Comment:
is `as any` the best we can do here?

Reply:
Looks like we don't need it, this was left over from dev